src/swsssdk/interface.py

```diff
@@ -366,4 +370,4 @@ def _connection_error_handler(self, db_name):
         logger.warning('Could not connect to Redis--waiting before trying again.')
         self.close(db_name)
         time.sleep(self.CONNECT_RETRY_WAIT_TIME)
-        self.connect(db_name, True)
+        self.connect(self.redis_db_map[db_name], db_name, True)
```

Comment:
Without a new dict, you can get `db_id` from `StrictRedis.connection_pool.connection_kwargs['db']`

Reply:
updated

Reply:
Offline discussed. My proposal is based on implementation details. So let's keep your code.